util/dynamicConfig.js

```diff
@@ -1,29 +1,47 @@
+/* eslint-disable no-param-reassign */
 const get = require("get-value");
 const unset = require("unset-value");
 
+function setDynamicConfigValue(event, value, config, field) {
+  value = value.replace("{{", "").replace("}}", "");
+  if (value.includes("||")) {
+    const path = value.split("||")[0].trim();
+    const getFieldVal = get(event, path);
+    if (getFieldVal) {
+      config[field] = getFieldVal;
+      unset(event, path);
+    } else {
+      config[field] = JSON.parse(value.split("||")[1].trim()).toString();
+    }
+  }
+}
+
 function getDynamicConfig(event) {
```

Comment:
@abandonned we need to recursively replace the dynamic config values.
CC @ruchiramoitra

Reply:
Sorry didn't get this. How do we want to use recursion? Currently our approach is to get the field and check for its type. If type and pattern matches we send it to dynamically Configure its value.

Reply:
yes discussed this with @abandonned

Reply:
Updated.